qiskit/providers/aer/openpulse/duffing_model_generators.py

```diff
@@ -0,0 +1,438 @@
+    drive strength r, which enter into the Hamiltonian model via the terms:
+        pi*(2*v-alpha)*O + pi*alpha*O*O + 2*pi*r*X*D(t),
+    where O is the number operator, X is the drive operator, and D(t) is the signal of the
+    corresponding DriveChannel.
+
+    Couplings between oscillators are specified in the argument coupling_dict, with keys being
+    edges, and values being the coupling strenghts. Couplings enter the Hamiltonian model via an
+    exchange coupling term:
+        2*pi*j*(A0*C1+C0*A1),
+    where j is the coupling strength, and A0*C1+C0*A1 is the exchange coupling operator between the
+    two oscillators, where A is the annihiliation operator, and C is the creation operator.
+
+    For each pair of coupled oscillators, pulse ControlChannels are prepared for doing cross
+    resonance (CR) drives between oscillators. A CR drive on oscillators 0 with target oscillators
+    1 is modeled with the Hamiltonian term:
+        2*pi*r0*X0*U(t),
+    where r0 is the drive strength for oscillators 0, X0 is the drive operator for oscillators 0,
+    and U(t) is the signal for the CR drive. U(t) is set to have carrier frequency equal to that of
+    the target oscillators.
+
+    Indices for ControlChannels corresponding to CR drives are provided in the returned cr_idx_dict.
+
+    Args:
+        num_oscillators (int): Number of oscillators in the model
+        dim_oscillators (int): Dimension of truncation for each oscillator
+        oscillator_freqs (list): Oscillator frequencies, assumed to be of length num_oscillators
+        anharm_freqs (list): Anharmonicity values, assumed to be of length num_oscillators
+        drive_strengths (list): Drive strength values, assumed to be of length num_oscillators
+        coupling_dict (dict): Specification of the coupling graph with keys being edges, and values
+                              the coupling strengths.
+
+                              For example:
+
+                              * ``{(0,1): 0.02, (1,3): 0.01}`` specifies a system in which
+                                oscillators (0,1) are coupled with strength 0.02, and (1,3) are
+                                coupled with strength 0.01
+        dt (float): Pixel size for pulse instructions
+        freq_symbol (str): Frequency symbol for internal HamiltonianModel
+        anharm_symbol (str): Anharmonicity symbol for internal HamiltonianModel
+        drive_symbol (str): Drive symbol for internal HamiltonianModel
+        coupling_symbol (str): Coupling symbol for internal HamiltonianModel
+
+    Returns:
+        tuple[PulseSystemModel, dict]: The generated oscillator system model, and a dict specifying
+                                       ControlChannel indices for CR driving.
+
+                                       For example:
+
+                                       * Given the coupling_dict ``{(0,1): 0.02, (1,3): 0.01}``,
+                                       the returned cr_idx_dict will be:
```

Comment:
If order matters for assigning channels, maybe the couping input should be a list `[[(i,j), val), ...]`, where list-position becomes channel number, rather than a dict

Reply:
The issue here is that for the purpose of specifying couplings there should be no difference between `(0, 1)` and `(1, 0)`; they should both just mean that qubits 0 and 1 are coupled. As such if we want the user to specify u channel orders themselves it should be in a separate argument (as otherwise `val` will need to be duplicated for both `(0,1)` and `(1, 0)`). In terms of method signature/output it would be simpler to make the user specify u channel orderings themselves, but in terms of actual use it will be more cumbersome.

It's somewhat annoying to deal with as for u channels, consistency in order matters, but the ordering itself is completely arbitrary.